src/environment.ts

```diff
@@ -20,12 +20,13 @@ function installLazyGlobals(env: SecureEnvironment, baseGlobalThis: object, base
             const { set: originalSetter, get: originalGetter, value: originalValue } = descriptor;
             if (!isUndefined(originalGetter)) {
                 descriptor.get = function get(): any {
-                    return env.getSecureValue(originalGetter.call(baseGlobalThis));
+                    const value: any = apply(originalGetter, baseGlobalThis, []);
+                    return env.getSecureValue(value);
                 };
             }
             if (!isUndefined(originalSetter)) {
                 descriptor.set = function set(v: any): void {
-                    originalSetter.call(baseGlobalThis, v);
+                    apply(originalSetter, baseGlobalThis, [v]);
```

Comment:
I prefer call here since it doesn't require the extra array for `v`, but maybe we can have something that does it.

Reply:
Yap. I had a `call` helper for a while that used `Reflect.apply` under the hood with a reused array. I can add that here too :)

Reply:
let's keep it like this, I will open an issue to track this separate.